Sources/Logging/Docs.docc/BestPractices/001-ChoosingLogLevels.md

```diff
@@ -128,11 +128,84 @@ logger.info("Request failed")
 ```swift
 // ❌ Bad: Normal operations at info level flood production logs
 logger.info("HTTP request received")
-logger.info("Database query executed") 
+logger.info("Database query executed")
 logger.info("Response sent")
 
 // ✅ Good: Use appropriate levels instead
 logger.debug("Processing request", metadata: ["path": "\(path)"])
 logger.trace("Query", metadata: ["sql": "\(query)"])
 logger.debug("Request completed", metadata: ["status": "\(status)"])
 ```
+
+### Disable log levels during compilation
+
+SwiftLog provides compile-time traits to eliminate less severe log levels from
+your binary reducing the runtime overhead.
+
+#### Motivation
+
+When deploying applications to production, you often know in advance which log
+levels will never be needed. For example, a production service might only need
+warning and above, while trace and debug levels are only useful during
+development. By using traits, you can completely remove these unnecessary log
+levels at compile time, achieving zero runtime overhead.
+
```

Comment:
We should mention here that these traits really should only be set by applications since they apply globally across the dependency tree. Libraries should not pass these traits.

Reply:
Added